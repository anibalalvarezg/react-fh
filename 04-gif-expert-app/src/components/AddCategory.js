import React, { useState } from 'react';

export const AddCategory = () => {

    const [inputValue, setInputValue] = useState('Hello World');

    const handleInputChange = (event) => {
        setInputValue(event.target.value)
    };

    const handleSubmit = (event) => {
        event.preventDefault(); 
        console.log('Submit done!');
    };

    return (
        <form onSubmit={ handleSubmit }>
            <h1>{ inputValue }</h1>
            <input
                type="text"
                value={ inputValue } 
                onChange={ handleInputChange } 
            />
        </form>
    );
}
