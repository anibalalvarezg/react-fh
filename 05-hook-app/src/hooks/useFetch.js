import { useEffect, useState } from "react"

export const useFetch = ( url ) => {
    const [state, setstate] = useState({ data: null, loading: true, error: null });

    useEffect(() => {
        setstate({ data: null, loading: true, error: null })
        fetch(url)
            .then(response => response.json())
            .then(data => {
                setstate({
                    data,
                    error: null,
                    loading: false,
                });
            })
    }, [url]);

    return state;
}
