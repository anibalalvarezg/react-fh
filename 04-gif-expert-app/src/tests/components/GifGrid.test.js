import '@testing-library/jest-dom';
import { shallow } from 'enzyme'; 
import React from 'react';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');
describe('Test GifGrid component', () => {
    
    const category = 'Rick and Morty';
    test('should display component correctly', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid  category={category} />); 
        expect( wrapper ).toMatchSnapshot();
    });
    
    test('shold display item when download images', () => {

        const gifs = [{
            id: 'ABC', 
            url: 'https://localhost/asd',
            title: 'anything'
        }]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid  category={category} />); 
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false ); 
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    });
    
});
