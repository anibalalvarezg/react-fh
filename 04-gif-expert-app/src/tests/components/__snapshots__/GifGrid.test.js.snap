// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifGrid component shold display item when download images 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Rick and Morty
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="anything"
      url="https://localhost/asd"
    />
  </div>
</Fragment>
`;

exports[`Test GifGrid component should display component correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Rick and Morty
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
